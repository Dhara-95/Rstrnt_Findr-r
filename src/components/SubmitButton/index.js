import React from "react";

const SubmitButton = (props) => {
  const { label, handleButtonClick, value } = props;

  return (
    <button
      className="submit-button rounded py-2 px-8 bg-teal hover:bg-teal-hover active:bg-teal-click focus:outline-none"
      type="button"
      value={value}
      onClick={() => handleButtonClick(value)}
    >
      {label}
    </button>
  );
};

export { SubmitButton };
