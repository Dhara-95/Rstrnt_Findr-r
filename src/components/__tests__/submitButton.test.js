import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";

import { SubmitButton } from "../SubmitButton";

it("displays the button with the correct props", () => {
  const tree = renderer
    .create(
      <SubmitButton
        label={"Find"}
        value={"value"}
        handleButtonClick={"function"}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("calls the passed in function on button click", async () => {
  const myMock = jest.fn();
  render(
    <SubmitButton label={"Find"} value={"value"} handleButtonClick={myMock} />
  );
  const button = screen.getByRole("button");
  userEvent.click(button);

  expect(myMock).toHaveBeenCalledTimes(1);
});
