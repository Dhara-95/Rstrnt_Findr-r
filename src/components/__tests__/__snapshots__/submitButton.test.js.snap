// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays the button with the correct props 1`] = `
<button
  className="submit-button rounded py-2 px-8 bg-teal hover:bg-teal-hover active:bg-teal-click focus:outline-none"
  onClick={[Function]}
  type="button"
  value="value"
>
  Find
</button>
`;
